### PR TITLE
feat(SDK-2004) : remove v1/logout network call

### PR DESCRIPTION
**Pull request by JagadeeshKaricherla-branch**
## Reference
SDK-2004 -- remove v1/logout network call

## Description
remove v1/logout network call inside logout public method

## Risk Assessment [`HIGH` || `MEDIUM` || `LOW`]

- [ ] I, the PR creator, have tested — integration, unit, or otherwise — this code.

## Reviewer Checklist (To be checked off by the reviewer only)

- [ ] JIRA Ticket is referenced in PR title.
- Correctness & Style
    - [ ] Conforms to [AOSP Style Guides](https://source.android.com/setup/contribute/code-style)
    - [ ] Mission critical pieces are documented in code and out of code as needed.
- [ ] Unit Tests reviewed and test issue sufficiently.
- [ ] Functionality was reviewed in QA independently by another engineer on the team.

cc @BranchMetrics/saas-sdk-devs for visibility.

### CODE REVIEW

File: Branch-SDK/src/main/java/io/branch/referral/Branch.java [hunk 1]
```diff
@@ -1232,9 +1232,15 @@ public void logout() {
...
     public void logout(LogoutStatusListener callback) {
-        ServerRequest req = new ServerRequestLogout(context_, callback);
-        if (!req.constructError_ && !req.handleErrors(context_)) {
-            handleNewRequest(req);
+        prefHelper_.setInstallParams(PrefHelper.NO_STRING_VALUE);
+        prefHelper_.setSessionParams(PrefHelper.NO_STRING_VALUE);
+        prefHelper_.setIdentity(PrefHelper.NO_STRING_VALUE);
+        prefHelper_.clearUserValues();
+        //On Logout clear the link cache and all pending requests
+        linkCache_.clear();
```
Comment (ahmednawar) [inline Branch-SDK/src/main/java/io/branch/referral/Branch.java:1240]: what does linkCache contain?

Comment (JagadeeshKaricherla-branch) [reply Branch-SDK/src/main/java/io/branch/referral/Branch.java:1240]: It caches link returned by create url ( generateShortLink)

Comment (JagadeeshKaricherla-branch) [reply Branch-SDK/src/main/java/io/branch/referral/Branch.java:1240]: This was done earlier like this :
image

Comment (NidhiDixit09) [reply Branch-SDK/src/main/java/io/branch/referral/Branch.java:1240]: @JagadeeshKaricherla-branch  If installParams are cleared from the preferences, how we will get them again ? If user logs out once, API firstReferringParams will always return null.

Comment (JagadeeshKaricherla-branch) [reply Branch-SDK/src/main/java/io/branch/referral/Branch.java:1240]: @NidhiDixit09  There is a check here for that use case 
https://github.com/BranchMetrics/android-branch-deep-linking-attribution/blob/c89fd4938276ee6b4a4a6cbef089688d40241ca7/Branch-SDK/src/main/java/io/branch/referral/Branch.java#L1453
Also, emptying install params was done earlier on API response, it got relocated now.
image

File: Branch-SDK/src/main/java/io/branch/referral/ServerRequestLogout.java [hunk 2]
```diff
@@ -1,100 +0,0 @@
...
-    }
-
-    public ServerRequestLogout(Defines.RequestPath requestPath, JSONObject post, Context context) {
-        super(requestPath, post, context);
-    }
-
-    @Override
-    public void onRequestSucceeded(ServerResponse resp, Branch branch) {
-        try {
-            prefHelper_.setSessionID(resp.getObject().getString(Defines.Jsonkey.SessionID.getKey()));
```
Comment (gdeluna-branch) [inline Branch-SDK/src/main/java/io/branch/referral/ServerRequestLogout.java:51]: Do these values also need to be cleared in `public void logout(LogoutStatusListener callback)`?

**Review by gdeluna-branch — CHANGES_REQUESTED** [review]

File: Branch-SDK/src/main/java/io/branch/referral/ServerRequestLogout.java (hunk 2 above)
Comment (JagadeeshKaricherla-branch) [reply Branch-SDK/src/main/java/io/branch/referral/ServerRequestLogout.java:51]: Among the values cleared( set to NO_STRING_VALUE), its decided not to clear any referrer related params like Install params, Session params.

**Review by gdeluna-branch — APPROVED** [review]

**Pull request closed without merging**